File: server/routes/api/scripts/CompanyFactory.py
import json
from Company import Company
from Facility import Facility
from bisect import bisect_left


class CompanyFactory(object):
    __facilities = json.load(open("../../../../oshub_data/facilities.json"))

    def getCompany(self, company_name):
        """
        @param company_name - name of a company
        @returns a Company object pertaining to the company, containing information about
                 the company's known manufacturing facilities IF the company is in the Open
                 Supply Hub directory
                 Otherwise, returns None.
        """
        # Binary search to check if the given company is in our directory
        companies = list(self.__facilities.keys())
        if companies[bisect_left(
                companies, company_name.lower())] != company_name.lower():
            return None
        return Company(
            company_name.lower(),
            [Facility(i) for i in self.__facilities[company_name.lower()]])


if __name__ == "__main__":
    c = CompanyFactory().getCompany("adidas")

File: server/routes/api/scripts/CompanyFactory.js
const Company = require("./Company")
const Facility = require("./Facility");
const fs = require("fs");

const CompanyFactory = class {
    static async getCompany(company_name){
        /*
        @param company_name - name of a company
        @returns a Company object pertaining to the company, containing information about
                 the company's known manufacturing facilities IF the company is in the Open
                 Supply Hub directory
                 Otherwise, returns undefined.
        */
        // TODO: Binary search to check if the given company is in our directory (currently just a linear search)
        if(this.facilities === undefined){
            this.facilities = JSON.parse(fs.readFileSync("../oshub_data/facilities.json", "utf8"));
        }
        if(!(company_name.toLowerCase() in this.facilities)){ 
            return undefined;
        }

        var companys_facilities = []
        for(var i = 0;i<this.facilities[company_name.toLowerCase()].length;i++){
            companys_facilities.push(new Facility(this.facilities[company_name.toLowerCase()][i]));
        }
        return new Company(company_name.toLowerCase(), companys_facilities);

        }
    }
CompanyFactory.facilities = undefined;
module.exports = CompanyFactory;